### PR TITLE
HCL2: allow calling env as input var default value

### LINKED ISSUE #10152: Packer HCL Environment Variables without PKR_VAR_
#### Description

When using CI/CD tools such as Jenkins or CodeBuild you may want to reference environment variables that are not managed by the user. The specific example that I am looking at is getting the Git Repo URL and Commit ID from AWS CodeBuild which are stored in `CODEBUILD_SOURCE_VERSION` and `CODEBUILD_SOURCE_REPO_URL`.

When doing the HCL2_Upgrade the input variables resulted in the following:

```
variable "git_commit" {
  type    = string
  default = "${var.CODEBUILD_SOURCE_VERSION}"
}

variable "git_repo" {
  type    = string
  default = "${var.CODEBUILD_SOURCE_REPO_URL}"
}
```
Under JSON this works with the following:
```
    "variables": {
        "git_repo": "{{ env `CODEBUILD_SOURCE_REPO_URL`}}",
        "git_commit": "{{env `CODEBUILD_SOURCE_VERSION`}}"
    }
```

#### Current Error on validation
```
Error: Variables not allowed

  on packer-windows-2016-amazon-ami.json.pkr.hcl line 21, in variable "git_commit":
  21:   default = "${var.CODEBUILD_SOURCE_VERSION}"

Variables may not be used here.

Error: Variables not allowed

  on packer-windows-2016-amazon-ami.json.pkr.hcl line 26, in variable "git_repo":
  26:   default = "${var.CODEBUILD_SOURCE_REPO_URL}"

Variables may not be used here.
```

#### Use Case(s)

Leverage CI/CD created Environment Variables

### LINKED ISSUE #10187: Better support of standard environment variables in HCL2
Please search the existing issues for relevant feature requests, and use the
reaction feature
(https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)
to add upvotes to pre-existing requests.

#### Community Note

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request.
If you are interested in working on this issue or have submitted a pull request, please leave a comment.

#### Description

HCL2 seemingly does not support using standard system environment variables without resetting them to `PKR_VAR_<NEW_ENV>`.

For example, with JSON you can do the following:
```
    "jenkins_build_user": "{{env `BUILD_USER_ID`}}",
```

With the template conversion, it creates the following:
```
variable "jenkins_build_user" {
  type    = string
  default = "${var.BUILD_USER_ID}"
}
```

Results in the following error:
```
Error: Variables not allowed

  on centos7-base.json.pkr.hcl line 51, in variable "jenkins_build_user":
  51:   default = "${var.BUILD_USER_ID}"
```

Variables may not be used here.
#### Use Case(s)

Being able to utilize standard env vars from your system without having to set some packer specific environment variable such as `PKR_VAR_BLAH` would be more than beneficial. 

#### Potential configuration

Potentially allow for a ${env.Foo} setting within the variable block, such as:
```
variable "jenkins_build_user" {
  type    = string
  default = "${env.BUILD_USER_ID}"
}
```

#### Potential References
https://discuss.hashicorp.com/t/hcl2-environment-variables/9290

### PR DESCRIPTION
**Pull request by azr**
close #10187 
close #10152

### CODE REVIEW

File: website/pages/docs/from-1.5/functions/contextual/env.mdx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+layout: docs
+page_title: env - Functions - Configuration Language
+sidebar_title: env
+description: The env function retrieves environment values for input variables.
+---
+
+# `env` Function
+
+`env` allows to get the value for an environment variable inside input
```
Comment (SwampDragons) [inline website/pages/docs/from-1.5/functions/contextual/env.mdx:10]:
```suggestion
`env` allows you to get the value for an environment variable inside input
```

File: website/pages/docs/from-1.5/functions/contextual/env.mdx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+variables only. This is the only function that is callable from a variable
+block.
+
+```hcl
+variable "aws_region" {
+  default = env("AWS_DEFAULT_REGION")
+}
+```
+
+In the previous example the value of `aws_region` will be what's behind the
```
Comment (SwampDragons) [inline website/pages/docs/from-1.5/functions/contextual/env.mdx:20]:
```suggestion
In the previous example, the value of `aws_region` will be what's stored in the
```

File: website/pages/docs/from-1.5/functions/contextual/env.mdx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+`AWS_DEFAULT_REGION` env var, unless aws_region is set in a [manner that takes
```
Comment (SwampDragons) [inline website/pages/docs/from-1.5/functions/contextual/env.mdx:21]:
```suggestion
`AWS_DEFAULT_REGION` env var, unless aws_region is also set in a [manner that takes
```

**Review by SwampDragons — APPROVED** [review]
Looks great, and works!

**Review by nywilken — APPROVED** [review]
Nice 👍

**Comment by azr** [comment]
Thanks for reviewing @SwampDragons & @nywilken :D

**Pull request merged**